src/main/java/net/rubygrapefruit/platform/file/FileWatcher.java

```diff
@@ -0,0 +1,20 @@
+package net.rubygrapefruit.platform.file;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+/**
+ * A handle for watching file system locations.
+ */
+public interface FileWatcher extends Closeable {
+    FileWatcher EMPTY = new FileWatcher() {
+        @Override
+        public void close() {}
+    };
+
+    /**
+     * Closes the watcher and releases any native resources.
+     */
+    @Override
+    void close() throws IOException;
```

Comment:
Should clarify what happens when `close()` is called and there are some pending notifications or the callback is currently being executed.

Reply:
Clarified.